### PR TITLE
Feature/inactive slot confirmation #145

### PR DESCRIPTION
**Pull request by va-dym-ko-val**
image
image
image
image
image
image
image

### CODE REVIEW

File: frontend/src/app/modules/availability/slot/slot.component.ts [hunk 1]
```diff
@@ -87,9 +93,49 @@ export class SlotComponent extends BaseComponent implements OnDestroy {
...
+    }
+
+    public changeSlotActivity() {
+        this.http
+            .updateSlotEnabling(this.slot?.id)
+            .pipe(this.untilThis)
+            .subscribe(
+                () => {
+                    this.notifications.showSuccessMessage('Slot`s activity was successfully changed');
+                    window.location.reload();
```
Comment (tatianahutii) [inline frontend/src/app/modules/availability/slot/slot.component.ts]: Why do we need this?

**Review by tatianahutii — CHANGES_REQUESTED** [review]

File: frontend/src/app/modules/availability/slot/slot.component.ts (hunk 1 above)
Comment (va-dym-ko-val) [reply frontend/src/app/modules/availability/slot/slot.component.ts]: Notification or reloading?

Comment (va-dym-ko-val) [reply frontend/src/app/modules/availability/slot/slot.component.ts]: image

Comment (tatianahutii) [reply frontend/src/app/modules/availability/slot/slot.component.ts]: reloading

**Review by tatianahutii — APPROVED** [review]

File: backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/AvailabilityService.cs [hunk 2]
```diff
@@ -152,6 +152,22 @@ public async Task<AvailabilitySlotDto> UpdateAvailabilitySlot(long id, SaveAvail
...
+                throw new KeyNotFoundException("Availability slot doesn't exist");
+            }
+            else
+            {
+                availabilitySlot.IsEnabled = !availabilitySlot.IsEnabled;
+                _context.AvailabilitySlots.Update(availabilitySlot);
+                await _context.SaveChangesAsync();
+                return !availabilitySlot.IsEnabled;
+            }
+        }
```
Comment (tatianahutii) [inline backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/AvailabilityService.cs]:
```suggestion
            var availabilitySlot = await _context.AvailabilitySlots.FirstOrDefaultAsync(slot => slot.Id == id) ?? throw new KeyNotFoundException("Availability slot doesn't exist");
            
              availabilitySlot.IsEnabled = !availabilitySlot.IsEnabled;
              _context.AvailabilitySlots.Update(availabilitySlot);
              await _context.SaveChangesAsync();
              return !availabilitySlot.IsEnabled;
        }
```

File: frontend/src/app/core/services/availability-slot.service.ts [hunk 3]
```diff
@@ -26,6 +27,10 @@ export class AvailabilitySlotService {
         return this.httpService.putRequest<IAvailabilitySlot>(`/availability/${slotId}`, updateAvailability);
     }
 
+    public updateSlotEnabling(slotId: bigint | undefined): Observable<boolean> {
+        return this.httpService.postRequest<boolean>(`${this.routePrefix}/enabling/${slotId}`, {});
```
Comment (tatianahutii) [inline frontend/src/app/core/services/availability-slot.service.ts]:
```suggestion
    public updateSlotEnabling(slotId: bigint | undefined) {
        return this.httpService.postRequest<boolean>(`${this.routePrefix}/enabling/${slotId}`, {});
```

File: frontend/src/app/core/services/availability-slot.service.ts [hunk 4]
```diff
@@ -2,6 +2,7 @@ import { Injectable } from '@angular/core';
 import { IAvailabilitySlot } from '@core/models/IAvailiabilitySlot';
 import { IUserPersonalAndTeamSlots } from '@core/models/IUserPersonalAndTeamSlots';
 import { ISaveAvailability } from '@core/models/save-availability-slot/ISaveAvailability';
+import { Observable } from 'rxjs';
```
Comment (tatianahutii) [inline frontend/src/app/core/services/availability-slot.service.ts]: Please delete this

**Review by tatianahutii — CHANGES_REQUESTED** [review]

**Review by apofis96 — APPROVED** [review]

**Review by tatianahutii — APPROVED** [review]

**Pull request merged**